### PR TITLE
[Bugfix] Zero-init ROCm MLA attention output buffers for graph padding

### PR DESCRIPTION
**Pull request by andyluo7**
## Summary
This ports the zero-init MLA output-buffer fix to the ROCm MLA backends.

Specifically:
- `vllm/v1/attention/backends/mla/rocm_aiter_mla.py`
- `vllm/v1/attention/backends/mla/rocm_aiter_mla_sparse.py`

The change replaces per-call uninitialized / freshly allocated decode outputs with reusable zero-initialized output buffers, matching the intent of the existing CUDA-side `#37442` fix.

## Root cause
Under graph capture / padded decode paths, unused MLA output rows can retain stale values if the backend does not guarantee they are written. On ROCm:
- dense AITER MLA was allocating a new `torch.zeros(...)` output every call, which was safe but suboptimal for graph replay
- sparse AITER MLA used `torch.empty(...)`, which could expose stale / garbage data in unwritten padded slots

## Fix
- add persistent `self._decode_out` buffers
- allocate them with `torch.zeros(...)`
- reuse via slicing (`self._decode_out[:B]` / `[:num_tokens]`)

## Validation
Validated on MI300X ROCm with forced `attention_backend='ROCM_AITER_MLA'` using `deepseek-ai/DeepSeek-V2-Lite`:
- confirmed log: `Using AITER MLA backend.`
- passed model load
- passed torch.compile
- passed KV cache init
- passed graph capture
- end-to-end run completed successfully

This is the ROCm counterpart to `#37442`.

### CODE REVIEW

**Comment by shantipriya-amd** [comment]
@tjtanaa The companion Quark fix is now in #39498. Both PRs are required for correct DeepSeek-V3.2-mxfp4 serving on ROCm — validation data (GSM8K 0.9515, throughput benchmarks, functional tests) included in both.

**Comment by stefanskiasan** [comment]
Independent validation on **MI350X (gfx950)**: this PR fixes a real, measurable correctness bug, not just a theoretical one.

**Path:** `ROCM_AITER_MLA_SPARSE` with fp8 KV cache and speculative decoding, full CUDA graphs.

**Symptom:** under concurrent load, a small fraction of responses break off mid-word and degenerate into digit soup. The token inserted at the break point is consistently **8191**:

```
'... ist **Tokiohandle,0,0.1,112. the0322 3n.0\nThe2222...'
'... the key differenceshandle097.9223,122. The91,3212...'
```

**Why 8191:** `VOCAB_BLOCK_SIZE = 8192` in the rejection sampler. Since #50183, NaN block maxima are mapped to `-inf` so `tl.argmax` stays in bounds. If *every* block is `-inf` (the whole logits row is NaN), block 0 wins, and its own NaN-corrupted local argmax is the last position: 8191. A single NaN in the hidden state therefore surfaces as a plausible-looking token, after which the state is corrupt. See #53029.

**Where the NaN comes from:** `_forward_mla` allocates its output with `torch.empty`. With CUDA graphs the query is padded to the graph size, and rows the kernel skips keep whatever was in memory — occasionally NaN/Inf. That is exactly what this PR addresses.

**Measurements** (same detector, same prompts, same instance, only the patch differs):

| configuration | requests | corrupted | rate |
|---|---|---|---|
| baseline | ~150 | 3 | 2.0 % |
| + QuickReduce bf16→fp16 cast disabled | 2120 | 16 | 0.75 % |
| + QuickReduce acquire fence | 880 | 3 | 0.34 % |
| **+ this PR (zero-init)** | **1040** | **0** | **0 %** |

At the previous 0.34 % rate, 3.5 failures would be expected in 1040 requests; observing zero gives p ≈ 0.03. Against the 0.75 % row, p < 0.001.

Two notes for anyone trying to reproduce it:
- It only appears **with speculative decoding**, because the rejection sampler is what turns a NaN row into a token. Without spec decode we saw 0/224 — which makes it easy to misattribute to the speculation method itself.
- It is rare enough that short runs prove nothing. We declared it fixed twice after ~200 requests and were wrong both times.
